src/app/root-store/server-data/vm-snapshots/vm-snapshots-entity-actions.selectors.ts

```diff
@@ -0,0 +1,48 @@
+import { createSelector } from '@ngrx/store';
+import * as vmSelectors from '../../../reducers/vm/redux/vm.reducers';
+import { EntityAction } from '../../../shared/interfaces';
+import { VmState } from '../../../vm/shared/vm.model';
+import * as vmSnapshotsActions from './vm-snapshots.actions';
+import * as vmSnapshotsSelectors from './vm-snapshots.selectors';
+
+export const getVmSnapshotEntityActionsEntities = createSelector(
+  vmSnapshotsSelectors.selectAll,
+  vmSelectors.selectEntities,
+  (vmSnapshots, virtualMachineEntities): { [id: string]: EntityAction[] } => {
+    return vmSnapshots.reduce((dictionary, vmSnapshot) => {
+      const vm = virtualMachineEntities[vmSnapshot.virtualmachineid];
+      const vmState = vm ? vm.state : undefined;
+      dictionary[vmSnapshot.id] = [
+        {
+          icon: 'mdi-camera',
+          text: 'SNAPSHOT_PAGE.ACTIONS.CREATE_SNAP_FROM_VM_SNAP',
+          disabled: false,
+          visible: false,
+          actionCreator: () =>
+            new vmSnapshotsActions.CreateVolumeSnapshot({ snapshotId: vmSnapshot.id }),
```

Comment:
The other two actions are named Revert and Delete. Is there any reason to have `CreateVolumeSnapshot` instead of `Create`?